index.d.ts

```diff
@@ -93,7 +93,9 @@ export function combineReducers<S>(reducers: ReducersMapObject): Reducer<S>;
  * transform, delay, ignore, or otherwise interpret actions or async actions
  * before passing them to the next middleware.
  */
-export type Dispatch = (action: any) => any;
+export interface Dispatch<S> {
+    <A extends Action>(action: A): A;
+}
```

Comment:
Having state type parameter `S` allows middlewares be more accurate in typings. For example, thunk function takes a dispatch and a `getState(): S`, so without `S` it couldn't be well-typed. See the example of this here below https://github.com/Igorbek/redux/blob/ts-def-improv/test/typescript/dispatch.ts#L9-L16

Regarding second question, I think this signature express exactly what initial (without midllewares) dispatch does. It take an action and returns it. If I used less generic signature `(action: Action) => Action`, due to TypeScript rules it would fail on object literals, such as `dispatch({ type: 'ADD_TODO', text: 'todo' })` (_Object literals may only specify known properties_). Moreover, it returns more precise result - typed action that was passed.